pkg/connection/reconcile_namespace.go

```diff
@@ -131,6 +151,42 @@ func (r *PulsarNamespaceReconciler) ReconcileNamespace(ctx context.Context, puls
 		BacklogQuotaType:            namespace.Spec.BacklogQuotaType,
 	}
 
+	if refs := namespace.Spec.GeoReplicationRefs; len(refs) != 0 {
+		for _, ref := range refs {
+			geoReplication := &v1alpha1.PulsarGeoReplication{}
+			namespacedName := types.NamespacedName{
+				Namespace: namespace.Namespace,
+				Name:      ref.Name,
+			}
+			if err := r.conn.client.Get(ctx, namespacedName, geoReplication); err != nil {
+				return err
+			}
+			log.V(1).Info("Found geo replication", "GEO Replication", geoReplication.Name)
+			destConnection := &resourcev1alpha1.PulsarConnection{}
+			namespacedName = types.NamespacedName{
+				Name:      geoReplication.Spec.DestinationConnectionRef.Name,
+				Namespace: geoReplication.Namespace,
+			}
+			if err := r.conn.client.Get(ctx, namespacedName, destConnection); err != nil {
+				log.Error(err, "Failed to get destination connection for geo replication")
+				if apierrors.IsNotFound(err) {
```

Comment:
Again, just asking we only want to return the error if its a not found error and proceed if its any other type of error or should this be `!apierrors.IsNotFound(err)`?

Reply:
Sorry, my bad, this is not needed.

Reply:
Fixed.